From: Dave Shield <D.T.Shield@liverpool.ac.uk>
Date: Tue, 28 Feb 2012 10:44:41 +0000 (+0000)
Subject: CHANGES: snmpd: BUGS: 3460364: Fix use of block factor when detecting error conditions
X-Git-Tag: v5.6.2.pre1~18
X-Git-Url: http://net-snmp.git.sourceforge.net/git/gitweb.cgi?p=net-snmp%2Fnet-snmp;a=commitdiff_plain;h=879bf7079d34fa46f6fcf54a01c8500beaece59a

CHANGES: snmpd: BUGS: 3460364: Fix use of block factor when detecting error conditions
---

diff --git a/agent/mibgroup/ucd-snmp/disk_hw.c b/agent/mibgroup/ucd-snmp/disk_hw.c
index ba26479..2b80476 100644
--- a/agent/mibgroup/ucd-snmp/disk_hw.c
+++ b/agent/mibgroup/ucd-snmp/disk_hw.c
@@ -389,8 +389,9 @@ tryAgain:
 
     case ERRORFLAG:
         long_ret = 0;
+        val = netsnmp_fsys_avail_ull(entry);
         if (( entry->minspace >= 0 ) &&
-            ( entry->avail < entry->minspace ))
+            ( val < entry->minspace ))
             long_ret = 1;
         else if (( entry->minpercent >= 0 ) &&
                  (_percent( entry->avail, entry->size ) < entry->minpercent ))
@@ -399,12 +400,13 @@ tryAgain:
 
     case ERRORMSG:
         errmsg[0] = 0;
+        val = netsnmp_fsys_avail_ull(entry);
         if (( entry->minspace >= 0 ) &&
-            ( entry->avail < entry->minspace ))
+            ( val < entry->minspace ))
                 snprintf(errmsg, sizeof(errmsg),
                         "%s: less than %d free (= %d)",
                         entry->path, entry->minspace,
-                        (int) entry->avail);
+                        (int) val);
         else if (( entry->minpercent >= 0 ) &&
                  (_percent( entry->avail, entry->size ) < entry->minpercent ))
                 snprintf(errmsg, sizeof(errmsg),
